Userland/Libraries/LibJS/Runtime/IteratorOperations.cpp

```diff
@@ -73,13 +73,38 @@ Object* iterator_next(Object& iterator, Value value)
 // 7.4.3 IteratorComplete ( iterResult ), https://tc39.es/ecma262/#sec-iteratorcomplete
 bool iterator_complete(GlobalObject& global_object, Object& iterator_result)
 {
-    return iterator_result.get(global_object.vm().names.done).value_or(Value(false)).to_boolean();
+    auto done = iterator_result.get(global_object.vm().names.done);
+    if (global_object.vm().exception())
+        return {};
+    return done.to_boolean();
```

Comment:
If the `get()` is guaranteed to never return an empty value (pre object rewrite), let's `VERIFY()` that before `to_boolean()`'ing it - otherwise let's keep the `value_or()` for now :)

also, for readability:

```suggestion
    auto& vm = global_object.vm();
    auto done = iterator_result.get(vm.names.done);
    if (vm.exception())
        return {};
    return done.to_boolean();
```

*apologies if this shows up twice somehow, I commented while the diff was updated and GitHub discarded the comment.*

Reply:
oh my bad, i wrote this in the Object rewrite and then backported it into a separate PR once i realised it was fixing non-object rewrite related issues as well, so i forgot to add those :sweat_smile: fixed! (Removal of the value_or is not included in the patch as a result btw, if you want to do that yourself :)